src/events/Windows.cpp

```diff
@@ -758,16 +759,28 @@ void Events::listener_commitWindow(void* owner, void* data) {
     if (!PWINDOW->m_pWorkspace->m_bVisible)
         return;
 
-    g_pHyprRenderer->damageSurface(PWINDOW->m_pWLSurface->resource(), PWINDOW->m_vRealPosition.goal().x, PWINDOW->m_vRealPosition.goal().y,
-                                   PWINDOW->m_bIsX11 ? 1.0 / PWINDOW->m_fX11SurfaceScaledBy : 1.0);
+    const auto PMONITOR = g_pCompositor->getMonitorFromID(PWINDOW->m_iMonitorID);
+
+    PMONITOR->debugLastPresentation(g_pSeatManager->isPointerFrameCommit ? "listener_commitWindow skip" : "listener_commitWindow");
+    if (g_pSeatManager->isPointerFrameCommit) {
+        g_pSeatManager->isPointerFrameSkipped = false;
+        g_pSeatManager->isPointerFrameCommit  = false;
+    } else
+        g_pHyprRenderer->damageSurface(PWINDOW->m_pWLSurface->resource(), PWINDOW->m_vRealPosition.goal().x, PWINDOW->m_vRealPosition.goal().y,
+                                       PWINDOW->m_bIsX11 ? 1.0 / PWINDOW->m_fX11SurfaceScaledBy : 1.0);
+
+    if (g_pSeatManager->isPointerFrameSkipped) {
```

Comment:
"frame" refers to a monitor frame? In that case renderer would be a better place for it? "frame" in seat means wl_pointer.frame

Reply:
`isPointerFrameSkipped` refers to `g_pSeatManager->sendPointerFrame();`

Reply:
ok good then